### PR TITLE
Install `git lfs` on infra (drone & chef)

### PR DESCRIPTION
**Pull request by snickell**
This PR installs the `git-lfs` binary on our infra in advance of starting to use git lfs to improve our repo size etc.
- Installs git-lfs on drone and chef instances
- We require git-lfs >= v3 in order to have support for ssh authentication (v2 only does https)
- Adds a chef integration test to ensure git-lfs is installed and >= v3
- Adds the ~/.gitconfig `[filter "lfs"]` stanza created by `git lfs install` (hooks will be added to the repo itself as a separate step, this just prepares the system to be generally "git lfs enabled")
- Updates `SETUP.md` with instructions on installing `git lfs` on Mac, Linux and Windows

### CODE REVIEW

**Comment by snickell** [comment]
- [x] Chef test to verify git-lfs is installed and version >= 3 passes
- [x] Drone passes
- [x] Verify git-lfs is installed on adhoc and .gitconfig is setup correctly to clone a test repo using LFS

File: .drone.yml
Comment (snickell) [inline .drone.yml]: Will edit these to be (probably) `1.8.2` and push to dockerhub `codedotorg/code-dot-org:1.8.2` once we're ready for merge.

File: .drone.yml
Comment (snickell) [inline .drone.yml]: Reminder to self to also sign the drone.yml again after updating docker tag to final version prior to merge

File: .drone.yml
Comment (snickell) [reply .drone.yml]: make that 1.9.2

File: cookbooks/cdo-github-access/.kitchen.yml
Comment (snickell) [inline cookbooks/cdo-github-access/.kitchen.yml]: update this to ubuntu 20.04

**Comment by sureshc** [comment]
Do we also need to update our `SETUP.md` to have engineers install git-lfs? We have 3 build systems (Chef, Docker, and the manual SETUP.md) and we typically have to update all 3 when adding a new binary dependency.

**Comment by snickell** [comment]
@sureshc I was figuring on a separate PR for updating SETUP.md, but since this one is blocked on Dockerfile in general having trouble building, don't mind bundling the SETUP.md change into this one too.

**Comment by snickell** [comment]
@sureshc I updated the SETUP.md in this branch, want to take a look and tell me what you think?

File: SETUP.md
Comment (snickell) [inline SETUP.md:15]: These steps have been moved below "Install OS-specific pre-requisites" because Git LFS should be installed before cloning.

File: .drone.yml
Comment (snickell) [reply .drone.yml]: make that 1.9.3 😅

Comment (Hamms) [reply .drone.yml]: :upside_down_face:

File: SETUP.md
Comment (Hamms) [inline SETUP.md:17]: Oh, thank you for catching and fixing this!

File: cookbooks/cdo-github-access/metadata.rb
Comment (Hamms) [inline cookbooks/cdo-github-access/metadata.rb:7]: Note you don't have to update this yourself; these version numbers [are determined programmatically](https://github.com/code-dot-org/code-dot-org/blob/staging/cookbooks/update_cookbook_versions), and [updates are committed as part of the staging build](https://github.com/code-dot-org/code-dot-org/commit/661abe389f6fc082e67a3e70fe1d962e85c87ed4).

That being said, I don't think it'll hurt anything to update them manually. Just want to make sure you know!

**Review by Hamms — APPROVED** [review]
Looks good to me! Super excited to get this landed

File: cookbooks/cdo-github-access/metadata.rb
Comment (snickell) [reply cookbooks/cdo-github-access/metadata.rb:7]: oh interesting, I couldn't get my local tests to pass until I upped this version, so maybe it was just a local issue

**Pull request merged**
